### PR TITLE
DashboardScene: Don't show switch to old dashboard architecture toggle unless you are in dev mode

### PR DESCRIPTION
**Pull request by torkelo**
* Make it possible to force using old architecture even when dashboardScene feature toggle is true
* Don't show `Switch to old dashboard page` unless your in dev mode. Hm.. might be a good fallback to have this as a visible option to help users work around bugs during first feature preview phase? 
* Refactor the conditions a bit in NavToolbarActions

### CODE REVIEW

File: public/app/features/dashboard-scene/scene/NavToolbarActions.tsx [hunk 1]
```diff
@@ -335,7 +334,7 @@ export function ToolbarActions({ dashboard }: Props) {
 
   toolbarActions.push({
     group: 'main-buttons',
-    condition: isEditing && !editview && !meta.isNew && !isViewingPanel && !isEditingPanel,
+    condition: isEditing && !meta.isNew && isShowingDashboard,
```
Comment (Sergej-Vlasov) [inline public/app/features/dashboard-scene/scene/NavToolbarActions.tsx:337]: if order doesn't matter can be 
```typescript
 condition: isEditingAndShowingDashboard && !meta.isNew
```

File: public/app/features/dashboard-scene/scene/NavToolbarActions.tsx [hunk 2]
```diff
@@ -316,7 +315,7 @@ export function ToolbarActions({ dashboard }: Props) {
 
   toolbarActions.push({
     group: 'settings',
-    condition: isEditing && dashboard.canEditDashboard() && !isViewingPanel && !isEditingPanel && !editview,
+    condition: isEditing && dashboard.canEditDashboard() && isShowingDashboard,
```
Comment (Sergej-Vlasov) [inline public/app/features/dashboard-scene/scene/NavToolbarActions.tsx:318]: if order doesn't matter can be 
```typescript
 condition: isEditingAndShowingDashboard && dashboard.canEditDashboard()
```

**Review by Sergej-Vlasov — APPROVED** [review]
> Don't show Switch to old dashboard page unless your in dev mode. Hm.. might be a good fallback to have this as a visible option to help users work around bugs during first feature preview phase?

Would agree to leave this toggle visible as a fallback for 1-2 months after release

**Review by dprokop — APPROVED** [review]
> Don't show Switch to old dashboard page unless your in dev mode. Hm.. might be a good fallback to have this as a visible option to help users work around bugs during first feature preview phase?

But this will reqiure some more work. The button has no effect when viewing dashboard with the `dashboardScene` toggle on, as the decision about the page rendered is happening upstream in the DAshboardPageProxy based on the feature toggle exclusively. So I don't think there's a point of showing this button unless we actually allow switching back to the old arch.

**Comment by torkelo** [comment]
@dprokop no it does work, you can now use scenes url flag to force use old architecture

**Comment by dprokop** [comment]
you are right @torkelo, i've missed the `?scenes=false`

**Pull request merged**